### DISCUSSION TITLE
Struggle but cannot reach webUI after new install

### DISCUSSION THREAD

**Discussion by AttiMM** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#466] [General]
Hi @visualapproach,
after not using our Maldives for 1,5 month, I've tried to upgrade from 3.5.1 to the latest 4.1.0 from scratch (because of big changes).
Following the newest Build instructions pdf, I did the following:
- connect PCB V1 to my computer via USB cable
- load the new source code in Visual Studio
- build & upload firmware via USB: SUCCESS, but I received some yellow lines, see the screenshots:
![image](https://user-images.githubusercontent.com/10764890/224965935-a5e46232-ff4c-40e2-bb7f-795baea9f79b.png)
![image](https://user-images.githubusercontent.com/10764890/224966133-b70e796d-787a-48a1-9039-d063d1646251.png)
![image](https://user-images.githubusercontent.com/10764890/224966202-4c777532-ac1c-4acc-9ebc-31da5c5cb2e0.png)
- build & upload Filesystem Image via USB: SUCCESS
![image](https://user-images.githubusercontent.com/10764890/224966586-d724de5b-0fab-4186-9e99-07feaeb94aba.png)

- leaving the USB connection to computer: I can see Lay-Z-Spa Module access point and can connect it with the given pw, but obtaining IP address was failured
![image](https://user-images.githubusercontent.com/10764890/224967822-c1dbeb63-033b-4e2a-93b2-615f1d8bbbf5.png)
- I couldn't reach http://layzspa.local (or the fixed IP address: 192.168.0.16 which was used before): I get empy page or ERR_ADDRESS_UNREACHABLE

Then I unconnected the module from PC and connected to the pump wires:

- Lay-Z-Spa Module access point is not showing in the list of APs
- I still can't reach the module web UI

- meanwhile Network Scanner can see and ping the module and see the ports opened:
![image](https://user-images.githubusercontent.com/10764890/224969839-a9599918-b268-4d68-84f5-29c042c42bf7.png)
![image](https://user-images.githubusercontent.com/10764890/224969903-a0bfa946-db37-4d8b-af5a-c6cd16f2ad78.png)
![image](https://user-images.githubusercontent.com/10764890/224969967-dffd807e-4d3b-489f-b304-153c2609b6cc.png)
![image](https://user-images.githubusercontent.com/10764890/224970047-505119aa-186c-4c92-818c-532b74d21c1d.png)

Connecting to DSP is the same: no sound and no displaying anything.

Could you help me what can be wrong and how can I fix it?
Thanks a lot,
Attila

**Comment by visualapproach** [comment]
Hi, the yellow warnings is nothing to worry about. It’s just telling the user of the libraries that we shouldn’t use certain old functions. To obtain the IP with the USB connected you need to start the serial monitor and it will print it in the console window. You should make the web ui work before connecting to pump because you have to select hardware config for your pump.

**Comment by AttiMM** [comment]
Thanks, but it didn't help. You can see the IP (same as I tried with) and a lot of strange chars. 
![image](https://user-images.githubusercontent.com/10764890/225082847-d710afc8-77e4-4534-8e3c-0c6b7cf7d26e.png)
After I started Serial monitor, Lay-Z-Spa Module access point disappeared in the list of APs.

So, I'm still in trouble.

**Reply by visualapproach** [reply]
You wrote: "leaving the USB connection to computer: I can see Lay-Z-Spa Module access point and can connect it with the given pw, but obtaining IP address was failured"
I'm not sure what you mean by that. Did you fill in the credentials for you Wi-Fi and click save?
In that case it should be good to go. It will not tell you anything about the IP. You have to look in the serial monitor, or the display, or as you did by scanning.
Either it's connected to a guest network or something, or the files didn't get uploaded right.
Visit this page:
http://192.168.0.16/dir/
What do you get?

**Reply by AttiMM** [reply]
To be honest, this part of pdf (and in your reply as well!) is not clear for me:
![image](https://user-images.githubusercontent.com/10764890/225391980-59b96a7c-29e7-4692-a681-fa764423026f.png)

As I wrote, I can see "Lay-Z-Spa Module" AP and could connect with pw "layzspam0dule". This connection was saved, it was trying obtaining IP address for a half a minute but finaly IP configuration failed message shown and it switched back to the usual AP of my wifi modem.
![image](https://user-images.githubusercontent.com/10764890/225390183-75d8d629-7ffe-45a0-8b2f-3aca0de4d605.png)

**Well, how / where should I log in and enter my wifi credentials? - this is unclear for me.**


Anyway this page is working now:
![image](https://user-images.githubusercontent.com/10764890/225390687-3465b1c1-9665-4b88-8ccf-a4de94ec52b9.png)

but http://192.168.0.16/ gives an empty page.

**Reply by visualapproach** [reply]
Now visit page 
http://192.168.0.16/update/
Let it work for several minutes and do not power off. It will update everything and upload the files.

**Reply by AttiMM** [reply]
![image](https://user-images.githubusercontent.com/10764890/225405396-e544602c-1f47-458e-bd6e-093668ea1e27.png)
I don't know whether it is count or not, but update.html is not found in dir list!


But in 2nd try I mistyped upload (instead of update) and got:
![image](https://user-images.githubusercontent.com/10764890/225405325-3f7b9e25-4c0d-4e5e-9df7-1b7dc7aa723c.png)

**Reply by visualapproach** [reply]
I didn't say .html and I even gave you a link to click

**Reply by visualapproach** [reply]
https://user-images.githubusercontent.com/10764890/225391980-59b96a7c-29e7-4692-a681-fa764423026f.png
It means you will be (should be) redirected to a webpage after connecting to that Wi-Fi access point. On that page you fill in ssid and password of your own Wi-Fi

**Comment by AttiMM** [comment]
**Yippiiii!!! It seems everything is working now!!!**

I'm sorry, I just typed update and forgot the "/". The ".html" part was added by the browser automatically.
But after I used your link with "update/" it was working for some minutes and finally I got everything fine!
Now I configure all the setting and try connect to the pump and display.

(But I still don't know what could be the problem...)

Thanks a lot for your patient and for your persistent help!

**Reply by visualapproach** [reply]
No problem. Don't know what was wrong but several files had the wrong filesize and the filelist.txt shouldn't be uploaded if done right but who knows. Glad it works now. Next time you want to update, there is a page for that in the ui so you don't need to struggle with platformio

**Reply by AttiMM** [reply]
Everything is fine with the pump and display as well! 
Yes, I know about the new update possibility! It will be much easier! :-)
I'm really thankful!

**Reply by visualapproach** [reply]
Thank you @AttiMM 🫲

**Comment by AttiMM** [comment]
Hi again,
it was too early reporting that everything is OK with display. Some strange thing:
- I can't hear the lovely melody after restart ESP (but displaying IP address and firmware version is OK)
- I can hear the tone up/down when target temp button is pressed, but the temperature in display always remains at actual temp (after a moment of blackout), and the target temp in webUI doesn't change from the last settings. It's like if display would be locked (but it's not). If I change the target temp in webUI I can see the changing in display as well, so I can manage it from webUI, but not with the display buttons.

**Reply by visualapproach** [reply]
> Hi again, it was too early reporting that everything is OK with display. Some strange thing:
> 
> * I can't hear the lovely melody after restart ESP (but displaying IP address and firmware version is OK)

That's because I haven't added the lovely melody yet in this version :-)

> * I can hear the tone up/down when target temp button is pressed, but the temperature in display always remains at actual temp (after a moment of blackout), and the target temp in webUI doesn't change from the last settings. It's like if display would be locked (but it's not). If I change the target temp in webUI I can see the changing in display as well, so I can manage it from webUI, but not with the display buttons.

Maybe pressing a whole second or more?

**Reply by AttiMM** [reply]
Unfortunatelly, it still not good, see:
[https://photos.app.goo.gl/hxABmxeN6FQXPP3P6](url)

**Reply by visualapproach** [reply]
link gave 404

**Reply by AttiMM** [reply]
try this:
https://photos.google.com/share/AF1QipPibATj2yp16ulM0GOadAzkCB5J0NLy4DjFCq1kFxGarS3YXv8pVojH7imuHf7RKg?key=c2ZVNW1KU0lYd01kNG5zUkpVUUVsSWxoZVFpV0lR

**Reply by visualapproach** [reply]
That worked thx. I'll think about it. Can you post a picture of your hardware config page:
![image](https://user-images.githubusercontent.com/22987250/225861432-221b3c78-c242-4031-bf76-87e2036009b7.png)

**Reply by AttiMM** [reply]
Sure:
![image](https://user-images.githubusercontent.com/10764890/225889744-a1b4af64-235b-4fa9-b5af-4c07a7016dc4.png)

**Reply by AttiMM** [reply]
Let you know that we have Maldives with hydrojets

**Reply by visualapproach** [reply]
Do the other buttons on the display work as expected?

**Reply by AttiMM** [reply]
Well, some of them yes - others not:
- Lock/Unlock: instead of the original short press needed to change, but maybe it's normal with this module.
- Timer: doesn't do anything (except beep) neither short nor long press
- Bubble yet: OK
- Hydrojet: OK
- Heating and Filter pump buttons: short press turns ON, but I can't turn it OFF neither short nor long press, only if turn off the unit with Main button
- C/F: OK
- Water target temp setting: explained above

Note: the unit is in the room now, I will put into operation in some weeks.

**Reply by visualapproach** [reply]
Okay. I assume it did work in v3.5.
I think what I'll do is to update master branch to 4.1.1. It makes it possible to update to the "beta" version. Then we can experiment a little on the beta.

**Reply by visualapproach** [reply]
Advice when you have updated to 4.1.1
It will probably take a while to come up with something but I'll tell you when I have something

**Reply by AttiMM** [reply]
> I assume it did work in v3.5.

Yes, perfectly.

Thanks!

**Reply by visualapproach** [reply]
ok, update to 4.1.1, then update again to BETA version.
Use chrome and on the home page right click and select "inspect"
Select the tab "console".
You will see a lot of text scrolling by. Press UP on the display and scroll up to where the change was. Should look like this:

![image](https://user-images.githubusercontent.com/22987250/226198590-2032b0aa-1ccf-4821-817e-52c22f078585.png)

**Reply by AttiMM** [reply]
I've updated to 4.1.1 (FW: "2023-03-19-0950") by OTA from webUI. 
Trying to update to BETA in the save way, but nothing has changed. 
Should I update to BETA version by wire with platformio? If yes: where can I find the source code for BETA version? I couldn't find in Releases.

Also, I can't find such an information like "from cio..." in console after pressing UP on display.

**Reply by visualapproach** [reply]
Hm something didn't go as expected. I'll fix another version later

**Reply by visualapproach** [reply]
If you update again it should work.

**Reply by AttiMM** [reply]
Something went really wrong after update from webUI: I cannot reach webUI now:
![image](https://user-images.githubusercontent.com/10764890/226538246-aa289ab1-6c09-4603-bae8-1ddef8173b4d.png)

After hard reset only the first part of local IP address (or nothing) is showing, then blackout, and after 2 secs I can see the new firmware version number with "1908" at the end.
Regardless the above, the display is working but with the same deficiency as before. 

I was trying update again with http://192.168.0.16/update/, but the result is same:
![image](https://user-images.githubusercontent.com/10764890/226537891-83469bcc-14a1-42b5-9968-e79d6a7384b6.png)

http://192.168.0.16/dir/:
bootlog.txt Size: 1013 Bytes remove
chkupdatefw.html.gz Size: 1452 Bytes remove
cmdq.json Size: 9 Bytes remove
config.html.gz Size: 3017 Bytes remove
debug.html.gz Size: 762 Bytes remove
favicon.ico Size: 9662 Bytes remove
favicon.png Size: 1817 Bytes remove
filelist.txt Size: 340 Bytes remove
function.js.gz Size: 491 Bytes remove
hwcfg.json Size: 69 Bytes remove
hwconfig.html.gz Size: 2454 Bytes remove
hwtest1.jpg Size: 10533 Bytes remove
hwtest2.jpg Size: 9432 Bytes remove
hwtestinfo.html.gz Size: 879 Bytes remove
index.html.gz Size: 2711 Bytes remove
index.js.gz Size: 3497 Bytes remove
main.css.gz Size: 1852 Bytes remove
manifest.json Size: 149 Bytes remove
menu.png Size: 223 Bytes remove
mqtt.html.gz Size: 1418 Bytes remove
remove.html.gz Size: 754 Bytes remove
settings.json Size: 291 Bytes remove
states.txt Size: 33 Bytes remove
success.html.gz Size: 671 Bytes remove
upload.html.gz Size: 829 Bytes remove
visualapproach.png Size: 8177 Bytes remove
webconfig.html.gz Size: 1510 Bytes remove
webconfig.json Size: 82 Bytes remove
wifi.html.gz Size: 2045 Bytes remove
wifi.json Size: 250 Bytes remove

**Reply by visualapproach** [reply]
Alright, you could upload the files again via usb (build filesystem + upload file system). Or if you're fed up with it there is a backup branch with the 3.5.1 version

**Reply by AttiMM** [reply]
I don't give up - I'm working as a tester now :)

**Reply by AttiMM** [reply]
Just a short report: 

- I installed firmware and files again with v4.1.0 via USB, then update OTA was succesful. 
- Connecting to the pump only: Module is available on my network, /dir is working, but webGUI is not.
- Connecting to the display too: Module is not available on my network.

At this point I'm going back to v3.5.1

**Reply by visualapproach** [reply]
Yes I understand. The most stable version is in the dev branch (beta). There was a bug that caused web server to respond super slow.
But it may not fix the display problem.

**Reply by AttiMM** [reply]
Thx, v4.0.0-beta seems to be good (display buttons are working well too).

And another report: many users (including me) has reported that Visual Studio sometimes cannot find the module on COM port when uploading filesystem image. Here is my experience: After uploading firmware, the first trial of uploading file system is OK. But after that all further attempts will be failed. Restarting computer will solve this problem every time. 
You may want to add this info to the instruction pdf.

**Reply by visualapproach** [reply]
Thanks for the heads up! Nice to hear it works now

**Reply by AttiMM** [reply]
> Try todays dev version. There was a memory bug in previous code before 2023-03-27

Do you advise me this dev version as a good base for a future update resolving my display buttons issue?

**Reply by visualapproach** [reply]
I advice that you install the latest dev version because it's better. It may not always be the case but today it is. I thought your display issue was solved? And by the way, the master is 4.1.1 not 4.0.0

**Comment by dasfliege** [comment]
I seem to have quite similar problems. I have installed everything from scratch and after i did that, i was perfectly able to access the integrated AP, configure my WiFi and MQTT. Everything is working fine except that i can't access the WebGUI now anymore. I have a ping running to the IP of the Webgui and everytime i try to access it, the ping stops for a moment. The WebGUI is displayed sometime, but it's not beeing displayed correctly. Looks like CSS is missing, so it's just a bunch of something. I tried updating it by accessing http://myip/update/ but that doesn't seem to do anything either. 

Any ideas?

**Reply by visualapproach** [reply]
What cio/dsp model did you configure?
Can you check if IP/info/ works?

**Reply by dasfliege** [reply]
I've configured 54173 for both, as i have the 2019 Maldives. But nothing is connected yet.

/info/ is working fine. But even accessing that breaks the ping for a second. It seems like every access to the Webserver is causing it to stop somehow. As the /info/ page is quite small, it seems to be fast enough to get the page befor the webserver stops.

This is what the /info/ is showing:
Stack size:          268 
Free Heap:           17920 
Min  Heap:           10904 
Core version:        3.1.1 
CPU fq:              160 MHz
Cycle count:         2622381286 
Free cont stack:     1816 
Sketch size:         699792 
Free sketch space:   2445312 
Max free block size: 16136

**Reply by visualapproach** [reply]
Thx I'll check later.

**Reply by dasfliege** [reply]
Tried to build with " build_type = debug" in order to get some more information on the serial console, but i get an error uploading it when this flag is set.

**Reply by visualapproach** [reply]
debug mode doesn't work for me either. And neither does project inspector which would make it easier to debug memory issues and such.

**Reply by visualapproach** [reply]
super annoying! I uploaded the same code that worked yesterday and got the same behaviour as you guys. Then I just changed platformio.ini a little and it worked again. And now it doesn't. I think it is a memory problem again. We're abusing the ESP to the limits and beyond...
I uploaded a new version (date 2023-03-21-1724). I hope it will work okayish until I fix it properly. No time to do it now.

**Reply by dasfliege** [reply]
Thanks man. I had a timeframe before where it was working flawlessly for like an hour or so but returned to having the same problems shortly after. However, even when the GUI was working just fine, i always had missing pings when clicking something in the GUI.

**Reply by visualapproach** [reply]
Try todays dev version. There was a memory bug in previous code before 2023-03-27

**Reply by dasfliege** [reply]
Is dev the same as the beta that i flash through the UI, or do i have to download the dev and flash via USB?
With the "2023-03-27-2004" i've installed today i still have heavy problems accesing the WebGUI.

EDIT:
Think i've answered the question myself :-) Just downloaded the dev branch and flashed via USB. Seems to be up to date now:
Current: 2023-03-31-1630
Master: 2023-03-27-2004
Beta: 2023-03-31-1630

**Reply by visualapproach** [reply]
It's the same

**Reply by dasfliege** [reply]
I still have heavy problems accessing the UI. If i flash everything from scratch (i have only set the SSID in my config.h, everything else is left as it is) it's working fine until i restart the ESP. So after flashing, i can perfectly access the UI and ping times to the IP are between 4-20ms. If i change anything in the HA or MQTT tab and reboot the ESP, the problem comes up. Accessing the UI is barely possible, pings are breaking up and if they work they are far above 200ms.
Also after the clean flash, the blue LED on the ESP blinks like it's doing something. After the first reboot the LED is never blinking again. Don't know if this has any meaning though...

Is there any need to like hard resettig the ESP prior to flashing a new software release? I have never done that, so that might be an issue if it is necessary?

**Reply by dasfliege** [reply]
Just did another test:
I have configured my mqtt settings in the config.h and performed another clean flash to the ESP. Everything was working fine. Low pings and the UI was fast and responsive. MQTT was working as well. I was able to restart the ESP either via the UI or by simply unplug it and it was working fine again after. Blue LED was always blinking.
I the went ahead and changed the pump model to 54173 and PCB to v2. After a reboot, the problem reappeared. No access to the UI, Pings are barely working and if they do, they are extremely high, the blue LED is not active at all. 

Could it be that there is an issue with just the 54173 model??

Just for information: I have not installed the PCB in the pump yet. So there is nothing connected to the pins. Does this has any impact?

**Reply by dasfliege** [reply]
Did some more testing:
The problem occurs as soon as i select the 54173 DSP. If i just select the 54173 CIO and PCB v2, the UI is still working perfectly. I do have the same problems when i select DSP 54154 or 54144. So my assuption is that there is a problem in the  DSP_4W.cpp. But i haven't tested all of the 4-Wire options though.

Would be interesting to know if i am the only one having problems with the 4-Wire versions??

**Reply by visualapproach** [reply]
@dasfliege Some comments: 
> Also after the clean flash, the blue LED on the ESP blinks like it's doing something. After the first reboot the LED is never blinking again. Don't know if this has any meaning though...

See the build instructions. Last page FAQ. 54173 uses other pinout compared to the default pre2021.

> Is there any need to like hard resettig the ESP prior to flashing a new software release? I have never done that, so that might be an issue if it is necessary?

No

> Just for information: I have not installed the PCB in the pump yet. So there is nothing connected to the pins. Does this has any impact?

No, I run a 8266 on the bench now with 2023-03-31-1630 and 54173 set, and no problem with web UI. Again, set everything from the Web UI instead of editing the code. 

> Could it be that there is an issue with just the 54173 model??

Sure, but not on the bench. Maybe your MQTT broker, router or something is the problem. First set hardware config from web UI. Then turn off MQTT and check after restart if the web UI is responsive. 
Also perform the hardware test from the web UI if possible. (If web UI is not working) After that you can enable MQTT and check if that's the problem.

**Comment by dasfliege** [comment]
Performed the following steps while my 8266 is laying in front of me on my table, only connected to USB, nothing else:

1. Extracted the latest dev version. Changed nothing in the code.
2. Flashed the 8266
3. Build Filesystem
4. Upload Filesystem
5. Reboot the 8266 by plugging USB out and in again
6. Connect via WiFi
7. Made sure that MQTT is disabled
8. In the GUI switched CIO from "6 wire, pre 2021" to "4 wire, 54173"
9. Restarted ESP -> GUI still working
10. In the GUI switched DSP from "6 wire, pre 2021" to "4 wire, 54173"
11. Restarted ESP -> GUI not working anymore, ping breaks up and blue LED not blinking anymore.

![image](https://user-images.githubusercontent.com/31273000/229591080-2114bc32-3595-4760-b27b-97657cdf1ebc.png)


I have not done the hardwaretest yet, as i first have to solder the JST-SM cables. In my understanding, i have to connect them to perform the hardware test?

**Reply by visualapproach** [reply]
can you post picture of main web page plz.
You (probably) don't need to do the hardware test if you haven't connected anything yet. I thought it might be some shorts or something.

**Reply by dasfliege** [reply]
This is how it looks after a clean flash without configuring anything:
![image](https://user-images.githubusercontent.com/31273000/229623525-e48a8213-9c51-4aa3-9944-3f64f242d18c.png)

**Reply by dasfliege** [reply]
This is after i have set CIO to 54173:
![image](https://user-images.githubusercontent.com/31273000/229624017-e2f109f3-d735-405c-9f71-ab7ff7020440.png)

**Reply by visualapproach** [reply]
Okay what if you 'update' to master via the web ui? Then update to beta via web ui. You have to do a downgrade to master first or it will not install the firmware with the same version. If you do it like this you will have an exact copy of my version which is working. If it still doesn't it should be something else

**Reply by dasfliege** [reply]
Still the same... 
I would love to do some troubleshooting myself to help to find the root cause, but there seem to be very limited possibilities to get any kind of direct console output or logs or something?

**Reply by visualapproach** [reply]
You can check console in browser (chrome: right click/inspect)

**Reply by dasfliege** [reply]
Well, that would require that the web UI is loaded. But if the webserver on the 8266 is crashing, i can't see anything on the browser console.

**Reply by dasfliege** [reply]
Are you using the exact same ESP as i do??

![Screenshot 2023-04-04 154709](https://user-images.githubusercontent.com/31273000/229813372-4f7eea0c-8ba3-40ec-89c3-27215250ee9e.jpg)

**Reply by visualapproach** [reply]
No I have 12E but it's mostly the antenna that differs according to the internet.

**Reply by dasfliege** [reply]
Yeah that's what i've read too. Anyway, i will order a new one. Don't know what else could be the problem, other than a hardware defect.

There's no working fork of this project for the ESP32 available right now, right? I would have a bunch of these here already...

**Reply by visualapproach** [reply]
nope. no 32 version now.

What happens in the output terminal (serial monitor) when you start and then switch to DSP54173 and restart from the web ui?
You can also uncomment the DDEBUG lines in platformio.ini and rebuild+upload to get more info in serial monitor.

**Reply by dasfliege** [reply]
This is the output after a clean flash:

```
Start
WiFi > Using WiFiManager Config Portal
*WM:
*WM: AutoConnect
*WM: Connecting as wifi client...
*WM: Status:
*WM: 7
*WM: Using last saved values, should be faster
*WM: Connection result: 
*WM: 3
*WM: IP Address:
*WM: 10.33.33.181
.
Failed to read mqtt.json. Using defaults.
10.33.33.181
End of setup()

```
This is the output after i changed to 54173 and restarted via the WebUI:
```

 ets Jan  8 2013,rst cause:2, boot mode:(3,7)

load 0x4010f000, len 3424, room 16
tail 0
chksum 0x2e
load 0x3fff20b8, len 40, room 8
tail 0
chksum 0x2b
csum 0x2b
v000a4050
~ld

Start
WiFi > Using WiFiManager Config Portal
*WM:
*WM: AutoConnect
*WM: Connecting as wifi client...
*WM: Status:
*WM: 7
*WM: Using last saved values, should be faster
*WM: Connection result: 
*WM: 3
*WM: IP Address:
*WM: 10.33.33.181
.
Failed to read mqtt.json. Using defaults.
10.33.33.181
End of setup()
```

This is the output after uncommenting all the DDEBUG lines, clean flash, switching to 54173 and restarte via Web UI:

```
 ets Jan  8 2013,rst cause:2, boot mode:(3,7)

load 0x4010f000, len 3424, room 16
tail 0
chksum 0x2e
load 0x3fff20b8, len 40, room 8
tail 0
chksum 0x2b
csum 0x2b
v000a5ae0
~ld

Start
WiFi > Using WiFiManager Config Portal
*WM:
*WM: AutoConnect
*WM: Connecting as wifi client...
*WM: Status:
*WM: 7
*WM: Using last saved values, should be faster
*WM: Connection result: 
*WM: 3
*WM: IP Address:
*WM: 10.33.33.181
.
Failed to read mqtt.json. Using defaults.
10.33.33.181
End of setup()
```

**Reply by visualapproach** [reply]
I tried using the autoportal instead of "specific AP" so that I got the *WM output also, but no difference. It just works. You can try using specific AP instead and see if it makes any changes.

**Reply by dasfliege** [reply]
I was already using specific AP but WM was turned on as well. I have now deactivated it. And for whatever magical reason, right now it seems to be working!
Blue LED does not flash anymore, but the UI is accessible and i loose much less pings then before. 

![image](https://user-images.githubusercontent.com/31273000/230584079-8a5cf310-e1da-4ac0-8a63-23949607ebe1.png)

ESP restart ...

 ets Jan  8 2013,rst cause:2, boot mode:(3,7)

load 0x4010f000, len 3424, room 16
tail 0
chksum 0x2e
load 0x3fff20b8, len 40, room 8
tail 0
chksum 0x2b
csum 0x2b
v000a5ae0
~ld

Start
.
Failed to read mqtt.json. Using defaults.
10.33.33.181
End of setup()

![image](https://user-images.githubusercontent.com/31273000/230585456-595b6e3a-3b2b-447a-99eb-41b3373dd6a7.png)

**Reply by visualapproach** [reply]
No. By setting it on the web page.  It ofc requires the web server to work but you said it did after new install

**Comment by dasfliege** [comment]
After installing Beta 1854 in order to fix the hydrojet-not-in-HA problem, i now have problems reaching the webserver again when the ESP is connected to the pump and display. It is working fine on the bench though, but sometimes needs 1-2 restarts of the board. As soon as it is connected to pump and dsp, i have the old issue with very high ping times or loosing pings completely most of the time. UI loads but is just a blank white screen. Any obvious reason for that? May i try the latest beta?

WiFi signal should be fine, as i have my pump right next to my bench

**Reply by visualapproach** [reply]
No obvious reason unfortunately. I don't think the beta will be better in this matter. Unless there is some random problem that goes away by uploading new firmware.

**Reply by visualapproach** [reply]
See #504
